pkg/event_processor/http_request.go

```diff
@@ -81,11 +81,10 @@ func (hr *HTTPRequest) detect(payload []byte) error {
 	//hr.Init()
 	rd := bytes.NewReader(payload)
 	buf := bufio.NewReader(rd)
-	req, err := http.ReadRequest(buf)
+	_, err := http.ReadRequest(buf)
```

Comment:
During the detect method, there is indeed no need to pass req to hr. This is because it will be reassigned in the Write method.